File: DatingAppClient/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams {
    gender: string;
    minimumAge = 18;
    maximumAge = 99;
    pageNumber = 1;
    pageSize = 12;
    orderBy = 'lastActivity';

    constructor(user: User) {
        this.gender = user.gender === 'masculino' ? 'feminino' : 'masculino';
    }
}

File: DatingApp.API/Helpers/UserParams.cs
namespace DatingApp.API.Helpers;

public class UserParams : PaginationParams
{
    public string CurrentUsername = string.Empty;
    public string? Gender { get; set; } = string.Empty;

    public int MinimumAge { get; set; } = 18;
    public int MaximumAge { get; set; } = 100;
    public string OrderBy { get; set; } = "lastActivity";
}
